src/backend.ts

```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  let keyEventFunction = (e: SystemKeyEvent) => {
+    const cancelDebouncedEvent = debounce(() => {
+      validKeyEvent = null;
+    }, 1000);
+    if (e.eKey === 0) {
+      // Have not found any race condition issues with this approach since the key event fires long before focus change
+      validKeyEvent = e;
+      cancelDebouncedEvent();
+    }
+  };
+  const unregisterSystemKeyEvents = () => {
+    keyEventFunction = () => {};
+  };
+  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => keyEventFunction(e));
+
   const { unregister: unregisterFocusChangeEvents } =
     SteamClient.System.UI.RegisterForFocusChangeEvents(
       throttle(async (fce: FocusChangeEvent) => {
         // don't try anything while an application is launching or it could pause it midlaunch
         if (appIsStartingUp) return;
         // skip if we already got such an event before
-        if (fce.focusedApp.pid === lastPid) return;
+        if (
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
```

Comment:
This whole logic should be merged with the Pause on focus loss feature, since the feature depends on `autoPause` being on. So after the `if (!(await loadSettings()).autoPause) return;`. This way the ToggleField also doesn't need to be aware of `overlayPause`.

So I think in order to merge all this functionality together we could do it like this (but please correct me if I missed something):

```diff
diff --git a/src/backend.ts b/src/backend.ts
index 662ee26..c25ce39 100644
--- a/src/backend.ts
+++ b/src/backend.ts
@@ -351,41 +351,14 @@ export function setupFocusChangeHandler(): () => void {
           return;
         lastPid = fce.focusedApp.pid;
         lastAppid = fce.focusedApp.appid;
-        // Only pause if the user has enabled it
-        if (
-          fce.focusedApp.appid === 769 &&
-          fce.focusedApp.strExeName !== "steamwebhelper" &&
-          (await loadSettings()).overlayPause
-        ) {
-          // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
-          if (validKeyEvent?.eKey === 0) {
-            validKeyEvent = null;
-            await Promise.all(
-              (Router.RunningApps as AppOverviewExt[]).map(async (a) => {
-                const appMD = await getAppMetaData(Number(a.appid));
-                // if the sticky pause state is on for this app don't do anything to it
-                if (appMD.sticky_state) {
-                  return a;
-                }
-                appMD.is_paused = await is_paused(appMD.instanceid);
-                // the app is not in focus
-                if (!appMD.is_paused) {
-                  appMD.is_paused = await pause(appMD.instanceid);
-                  appMD.pause_state_callbacks.forEach((cb) =>
-                    cb(appMD.is_paused)
-                  );
-                }
-                return a;
-              })
-            );
-            return;
-          } else {
-            return;
-          }
-        }
-
         if (!(await loadSettings()).autoPause) return;
         // AppID 769 is the Steam Overlay or a non-steam app
+        // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
+        const overlayPause =
+          fce.focusedApp.appid === 769 &&
+          fce.focusedApp.strExeName !== "steamwebhelper" &&
+          validKeyEvent?.eKey === 0 &&
+          (await loadSettings()).overlayPause;
         if (!fce.focusedApp.appid || fce.focusedApp.appid === 769) {
           const appid = await appid_from_pid(fce.focusedApp.pid);
           if (appid) {
@@ -409,7 +382,7 @@ export function setupFocusChangeHandler(): () => void {
               return a;
             }
             appMD.is_paused = await is_paused(appMD.instanceid);
-            if (appMD.instanceid === fce.focusedApp.pid) {
+            if (!overlayPause && appMD.instanceid === fce.focusedApp.pid) {
               // this is the focused app
               if (appMD.is_paused) {
                 appMD.is_paused = !(await resume(appMD.instanceid));
@@ -418,7 +391,7 @@ export function setupFocusChangeHandler(): () => void {
                 );
               }
             } else {
-              // the app is not in focus
+              // the app is not in focus or the overlay is on
               if (!appMD.is_paused) {
                 appMD.is_paused = await pause(appMD.instanceid);
                 appMD.pause_state_callbacks.forEach((cb) =>

```

The whole `RegisterForFocusChangeEvents` section would look like this:

```ts
  const { unregister: unregisterFocusChangeEvents } =
    SteamClient.System.UI.RegisterForFocusChangeEvents(
      throttle(async (fce: FocusChangeEvent) => {
        // don't try anything while an application is launching or it could pause it midlaunch
        if (appIsStartingUp) return;
        // skip if we already got such an event before
        if (
          fce.focusedApp.pid === lastPid &&
          fce.focusedApp.appid === lastAppid
        )
          return;
        lastPid = fce.focusedApp.pid;
        lastAppid = fce.focusedApp.appid;
        if (!(await loadSettings()).autoPause) return;
        // AppID 769 is the Steam Overlay or a non-steam app
        // Key event must be if the user pressed the 'STEAM' button. Don't pause for other buttons
        const overlayPause =
          fce.focusedApp.appid === 769 &&
          fce.focusedApp.strExeName !== "steamwebhelper" &&
          validKeyEvent?.eKey === 0 &&
          (await loadSettings()).overlayPause;
        if (!fce.focusedApp.appid || fce.focusedApp.appid === 769) {
          const appid = await appid_from_pid(fce.focusedApp.pid);
          if (appid) {
            fce.focusedApp.appid = appid;
          } else {
            return;
          }
        }
        // fce.focusedApp.pid is not the pid of the reaper but of the first child
        const pid = await pid_from_appid(fce.focusedApp.appid);
        if (pid) {
          fce.focusedApp.pid = pid;
        } else {
          return;
        }
        await Promise.all(
          (Router.RunningApps as AppOverviewExt[]).map(async (a) => {
            const appMD = await getAppMetaData(Number(a.appid));
            // if the sticky pause state is on for this app don't do anything to it
            if (appMD.sticky_state) {
              return a;
            }
            appMD.is_paused = await is_paused(appMD.instanceid);
            if (!overlayPause && appMD.instanceid === fce.focusedApp.pid) {
              // this is the focused app
              if (appMD.is_paused) {
                appMD.is_paused = !(await resume(appMD.instanceid));
                appMD.pause_state_callbacks.forEach((cb) =>
                  cb(appMD.is_paused)
                );
              }
            } else {
              // the app is not in focus or the overlay is on
              if (!appMD.is_paused) {
                appMD.is_paused = await pause(appMD.instanceid);
                appMD.pause_state_callbacks.forEach((cb) =>
                  cb(appMD.is_paused)
                );
              }
            }
            return a;
          })
        );
      }, 500)
    );
```